packages/ui/nodes/image/src/ImageElement/ImageResizable.tsx

```diff
@@ -0,0 +1,100 @@
+import React, { useCallback, useEffect } from 'react';
+import {
+  findNodePath,
+  select,
+  setNodes,
+  useEditorRef,
+} from '@udecode/plate-core';
+import { TImageElement } from '@udecode/plate-image';
+import { Resizable, ResizableProps } from 're-resizable';
+import { useReadOnly } from 'slate-react';
+import { createComponentAs } from '../utils/createComponentAs';
+import { useElement } from '../utils/useElement';
+import { imageWidthAtom, useImageAtom } from './imageAtoms';
+
+export interface ImageResizableProps extends ResizableProps {
+  /**
+   * Image alignment.
+   */
+  align?: 'left' | 'center' | 'right';
+
+  ignoreReadOnly?: boolean;
+}
+
+export const useImageResizable = ({
+  ignoreReadOnly,
+  align = 'center',
+  ...props
+}: ImageResizableProps): ResizableProps => {
+  const element = useElement<TImageElement>();
+  const editor = useEditorRef();
+  const readOnly = useReadOnly();
+
+  const { width: nodeWidth = '100%' } = element ?? {};
+
+  const [width, setWidth] = useImageAtom(imageWidthAtom);
```

Comment:
I am not sure I understand how the Atoms are being used here vs regular useState.